grakn-graph/src/test/java/ai/grakn/graph/internal/GraknGraphPropertyIT.java

```diff
@@ -581,11 +578,22 @@ public void whenCallingGetRuleType_TheResultIsTheSameAsGetType(@Open GraknGraph
         assertSameResult(() -> graph.getType(typeName), () -> graph.getRuleType(typeName.getValue()));
     }
 
-    @Ignore // TODO: Fix this test
     @Property
     public void whenCallingGetRelationAndTheRelationExists_ReturnThatRelation(
             @Open GraknGraph graph, @FromGraph Relation relation) {
-        assertEquals(relation, graph.getRelation(relation.type(), relation.rolePlayers()));
+        //Cannot compare against the exact relation because it is possible to temporarily create (within a transaction)
+        // duplicate relations. In this case it was creating 2 relations with no roles and roleplayers of the same type
+        // and returning one of them which is valid but may not be the one you are comparing against. Hence why the
+        // comparison is more defined.
+
+        Relation foundRelation = graph.getRelation(relation.type(), relation.rolePlayers());
+        if(foundRelation.getId().equals(relation.getId())){
+            assertEquals(relation, foundRelation);
+        } else { //This is possible when we have created duplicate empty relations. So we check everything we can.
+            assertThat(relation.rolePlayers().keySet(), containsInAnyOrder(foundRelation.rolePlayers().keySet()));
+            assertThat(relation.rolePlayers().values(), containsInAnyOrder(foundRelation.rolePlayers().values()));
+            assertEquals(relation.type(), foundRelation.type());
+        }
```

Comment:
I might look into a way to split this into two tests, since it feels like two cases.

Alternative more radical idea: this odd behaviour might be a hint that we shouldn't have this method at all, or that it should return all matching relations.